src/data/list/cycle.lean

```diff
@@ -430,73 +430,90 @@ instance : has_coe (list α) (cycle α) := ⟨quot.mk _⟩
 @[simp] lemma coe_eq_coe {l₁ l₂ : list α} : (l₁ : cycle α) = l₂ ↔ (l₁ ~r l₂) :=
 @quotient.eq _ (is_rotated.setoid _) _ _
 
-@[simp] lemma mk_eq_coe (l : list α) :
-  quot.mk _ l = (l : cycle α) := rfl
+@[simp] lemma mk_eq_coe (l : list α) : quot.mk _ l = (l : cycle α) :=
+rfl
 
-@[simp] lemma mk'_eq_coe (l : list α) :
-  quotient.mk' l = (l : cycle α) := rfl
+@[simp] lemma mk'_eq_coe (l : list α) : quotient.mk' l = (l : cycle α) :=
+rfl
 
-instance : inhabited (cycle α) := ⟨(([] : list α) : cycle α)⟩
+/-- The unique empty cycle. -/
+def nil : cycle α := ↑([] : list α)
 
-/--
-For `x : α`, `s : cycle α`, `x ∈ s` indicates that `x` occurs at least once in `s`.
--/
+@[simp] lemma coe_nil : ↑([] : list α) = @nil α :=
+rfl
+
+@[simp] lemma coe_eq_nil (l : list α) : (l : cycle α) = nil ↔ l = [] :=
+coe_eq_coe.trans is_rotated_nil_iff
+
+instance : has_emptyc (cycle α) := ⟨nil⟩
+
+@[simp] lemma empty_eq : ∅ = @nil α :=
+rfl
```

Comment:
It's not clear to me whether we want `∅` to be permitted notation for a cycle here.

Reply:
I'll let @pechersky decide, since I think `cycle` was their idea.

Reply:
My train of thought was, `list` also has the empty set notation even though it's universally discarded in favor of `[]`, so maybe there's a good reason behind it and the same should be true for `cycle`. Otherwise, I can remove this.

Reply:
```suggestion
/-- To match `list.has_emptyc`. -/
instance : has_emptyc (cycle α) := ⟨nil⟩

@[simp] lemma empty_eq : ∅ = @nil α :=
rfl
```

Reply:
That's a good enough argument for me. Let's just record the rationale.

bors d+